qiskit/compiler/transpile.py

```diff
@@ -206,10 +222,20 @@ def callback_func(**kwargs):
         config = user_config.get_config()
         optimization_level = config.get('transpile_optimization_level', 1)
 
+    if scheduling_method is None:
+        if any_delay_in(circuits):
+            raise TranspilerError("Invalid arguments: When transpiling circuits with delays, "
+                                  "'scheduling_method' is required.")
+    else:
+        if backend is None and not (basis_gates and instruction_durations):
+            raise TranspilerError("Invalid arguments: When scheduling circuits without backend,"
+                                  " 'basis_gates' and 'instruction_durations' are required.")
```

Comment:
basis_gates shouldn't be a requirement. should just try to schedule a circuit with the gates that exist. if one of the durations is unknown, raise error.